components/event-processor-manager/org.wso2.carbon.event.processor.manager.core/src/main/java/org/wso2/carbon/event/processor/manager/core/internal/CarbonEventManagementService.java

```diff
@@ -163,8 +163,12 @@ public void init(ConfigurationContextService configurationContextService) {
         executorService.schedule(new Runnable() {
             @Override
             public void run() {
-                log.info("Starting polling event adapters");
-                getEventReceiverManagementService().startPolling();
+                try {
+                    log.info("Starting polling event adapters");
+                    getEventReceiverManagementService().startPolling();
+                } catch (Exception e) {
```

Comment:
Any requirement to catch generic Exception? If not please only catch specific exception.